### ISSUE TITLE
Anonymous class indentation is wrongly reported as incorrect in some cases

### ISSUE THREAD

**Issue by valgog** [checkstyle/checkstyle#2829]
When trying to check the code, formatted according to the Google Java Code Style, I see some strange behaviour of `checkstyle`. Here an example of the code:

```java
package example;

/**
 * Inner class formatting identified by checkstyle wrongly.
 */
public class CheckstyleProblem {

  public interface Watcher {
    public void process(CheckstyleProblem problem);
  }

  public CheckstyleProblem(String name, int intArgument, Watcher watcher) {
    // bla bla
  }

  /**
   * This is an example call that shows how inner class instance.
   *
   * @return checkstyle problem
   */
  public CheckstyleProblem init() {
    // some example code that gets a Watcher

    CheckstyleProblem result;
    result =
        new CheckstyleProblem(
            "problematic code formatting?",
            3334,
            //
            // here checkstyle considers the indentation is wrong
            //
            new Watcher() {
              @Override
              public void process(CheckstyleProblem problem) {}
            });

    return new CheckstyleProblem(
        "code formatting is ok from the point of view of checkstyle",
        3334,
        //
        // here no warning from checkstyle
        //
        new Watcher() {
          @Override
          public void process(CheckstyleProblem problem) {}
        });
  }
}
``` 

Here are the error messages:

```text
[ant:checkstyle] [WARN] CheckstyleProblem.java:33: 'method def modifier' have incorrect indentation level 14, expected level should be one of the following: 8, 10, 12. [Indentation]
[ant:checkstyle] [WARN] CheckstyleProblem.java:35: 'object def rcurly' have incorrect indentation level 12, expected level should be one of the following: 6, 8. [Indentation]
```

**Comment by romani** [comment]
thanks for reporting, please change description to contain all required info for bug reporting, skip through http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug

**Comment by valgog** [comment]
@romani I have updated the description as requested.

**Comment by romani** [comment]
thanks a lot for good report, try to find workaround, issues of Indentation Check are not simple to resolve :( (time consuming and no enough people). You are welcome to contribute a fix, I will help you with development process.

**Comment by waschmittel** [comment]
Is there anything done to fix this? I currently cannot check indentation at all because of this.

**Comment by rnveach** [comment]
As long as the issue is open and no PRs have been started, then it is safe to assume nothing has been done.

**Comment by SButterfly** [comment]
Up this bug. Still having a problem
```java
    @Test
    void test() {
        var map = new HashMap<>() {{
            put(KEY1, VALUE1);
            put(KEY2, VALUE2);
        }};
    }
```

and checkstyle.xml
```xml
        <module name="Indentation">
            <property name="basicOffset" value="4"/>
            <property name="braceAdjustment" value="2"/>
            <property name="caseIndent" value="4"/>
            <property name="throwsIndent" value="4"/>
            <property name="lineWrappingIndentation" value="4"/>
            <property name="arrayInitIndent" value="2"/>
        </module>
```
